pkg/apis/work/v1alpha2/binding_types_helper.go

```diff
@@ -40,3 +40,35 @@ func (s *ResourceBindingSpec) RemoveCluster(name string) {
 
 	s.Clusters = append(s.Clusters[:i], s.Clusters[i+1:]...)
 }
+
+// GracefulEvictCluster removes specific cluster from the target list in a graceful way by
+// building a graceful eviction task.
+// This function no-opts if cluster not exist.
+func (s *ResourceBindingSpec) GracefulEvictCluster(name string, producer string, reason string, message string) {
+	// find the cluster index
+	var i int
+	for i = 0; i < len(s.Clusters); i++ {
+		if s.Clusters[i].Name == name {
+			break
+		}
+	}
+	// not found, do nothing
+	if i >= len(s.Clusters) {
+		return
+	}
+
+	// build eviction task
+	evictingCluster := s.Clusters[i].DeepCopy()
+	evictionTask := GracefulEvictionTask{
+		FromCluster: evictingCluster.Name,
+		Reason:      reason,
+		Message:     message,
+		Producer:    producer,
+	}
+	if evictingCluster.Replicas > 0 {
+		evictionTask.Replicas = &evictingCluster.Replicas
```

Comment:
If `evictingCluster.Replicas > 0`, we evict this workload by scaling down its replicas to 0 or deleting itself?

Reply:
All tasks in `graceful eviction tasks` will be removed eventually, which means to delete it, it's the same behavior with taint manager, `graceful eviction` essentially delayed the deleting process.

Reply:
OK. I guess if scheduling result moved from `spec.clusters` to `spec.gracefulEvictionTasks`, we don't acutually remove the replicas immediately, right? So is there another change about how to divide replicas considering `spec.gracefulEvictionTasks`?

Reply:
Now we divide replicas only by `spec.clusters`

Reply:
> I guess if scheduling result moved from spec.clusters to spec.gracefulEvictionTasks, we don't acutually remove the replicas immediately, right?

Yes, exactly. 

> So is there another change about how to divide replicas considering spec.gracefulEvictionTasks?

I guess no. From the scheduler's point of view, once a cluster has been evicted, it's gone even though it still keeps in `graceful eviction tasks`.

Reply:
I have figured it out through https://github.com/karmada-io/karmada/pull/2319#discussion_r940979768, the work will be deleted by `removeOrphanWorks`.

Reply:
Yeah, thanks.